### PR TITLE
Mirror DirectionalHints in RTL layouts

### LINKED ISSUE #2278: Callout/ContextualMenu: DirectionalHint should flip in RTL layouts
### Bug Report

- __Package version(s)__: 4.21.4

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

Products/sites affected: Outlook Web

#### Describe the issue:

Rather than forcing every developer to check for RTL layouts and switch DirectionalHints, this should be the default behavior of the control. *Not* swapping for RTL is the exception, not the general case.

### PR DESCRIPTION
**Pull request by MLoughry**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #2278
- [x] Include a change request file using `$ npm run change`

#### Description of changes

All code which uses DirectionalHints now mirrors them in RTL layouts by default. All such components now also have `directionalHintForRtl` properties to use as an escape hatch to avoid this default behavior.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by MLoughry** [comment]
@cschleiden, @dzearing wanted you to be aware of this change in case it affects any existing workarounds you have in your code.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts [hunk 1]
```diff
@@ -48,6 +48,12 @@ export interface IContextualMenuProps extends React.Props<ContextualMenu>, IWith
   directionalHint?: DirectionalHint;
 
   /**
+   * How the element should be positioned in RTL layouts.
+   * If not specified, a mirror of `directionalHint` will be used instead
+   */
+  directionalHintForRtl?: DirectionalHint;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts]: maybe we should just name this `overrideRTLFlip` or `forceLTRDirection` ? Having a hard time understanding `directionalHintForRtl`

File: packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Directional.Example.tsx [hunk 2]
```diff
@@ -6,11 +6,14 @@ import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
 import { Slider } from 'office-ui-fabric-react/lib/Slider';
 import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import { autobind } from 'office-ui-fabric-react/lib/Utilities';
+import { getRTL } from '@uifabric/utilities';
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/examples/ContextualMenu.Directional.Example.tsx]: The utilities import is right above (see autobind)

File: packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { CalloutContent } from './CalloutContent';
 import { DirectionalHint } from '../../common/DirectionalHint';
 import {
   IPoint,
-  IRectangle
+  IRectangle,
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts]: nit

File: packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts [hunk 4]
```diff
@@ -32,6 +32,12 @@ export interface ICalloutProps extends React.Props<Callout | CalloutContent> {
   directionalHint?: DirectionalHint;
 
   /**
+   * How the element should be positioned in RTL layouts.
+   * If not specified, a mirror of `directionalHint` will be used instead
+   */
+  directionalHintForRtl?: DirectionalHint;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts]: is RTL TitleCased or CAPS?

i see `getRTL`, maybe we should be consistent.

https://github.com/airbnb/javascript#naming--Acronyms-and-Initialisms

We need to get the TS coding guidelines moved into the fabric ghdocs folder.

**Comment by dzearing** [comment]
please dont merge until @MLoughry has had a chance to address changes

**Comment by cschleiden** [comment]
We don't use RTL in VSTS for now, so I think we should be good.

File: packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts (hunk 3 above)
Comment (MLoughry) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts]: Not sure what the nit is of?

**Pull request merged**